(Re)introduce Property.replace (#27986)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

It is possible to persist the preference by setting the `org.gradle.buildinit.comments` property to `false`.

<a name="replace-method"></a>
#### Updating lazy property based on its current value with `replace()`

[Lazy configuration](userguide/lazy_configuration.html) delays calculating a property’s value until it is required for the build.
Sometimes it is necessary to modify the property based on its current value, for example, by appending something to it.

> Previously, the only way to do that was to obtain the current value explicitly by calling `Property.get()`:

Reviewer: ```suggestion
Previously, the only way to do that was to obtain the current value explicitly by calling `Property.get()`:

```

```
val property = objects.property<String>()
property.set("some value")
property.set("${property.get()} and more" })

println(property.get()) // "some value and more""
```

> This could lead to performance issues like configuration cache misses.

Reviewer: ```suggestion

This could lead to performance issues like configuration cache misses.
```

> Trying to build the value in a lazy manner, for example, by using `property.set(property.map { "$it and more" })`, causes build failure because of a circular reference evaluation.

Reviewer: ```suggestion
Trying to build the value lazily, for example, by using `property.set(property.map { "$it and more" })`, causes build failure because of a circular reference evaluation.
```

> [`Property`](javadoc/org/gradle/api/provider/Property.html#update-org.gradle.api.Transformer-) and [`ConfigurableFileCollection`](javadoc/org/gradle/api/file/ConfigurableFileCollection.html#update-org.gradle.api.Transformer-) now provide their respective `replace(Transformer<...>)` methods that allow lazily building the new value based on the current one:

Reviewer 2: Links still refer to the and `#update...` anchor

```
val property = objects.property<String>()
property.set("some value")
property.replace { it.map { "$it and more" } }

println(property.get()) // "some value and more"
```

> Refer to the javadoc for [`Property.replace(Transformer<>)`](javadoc/org/gradle/api/provider/Property.html#replace-org.gradle.api.Transformer-) and [`ConfigurableFileCollection.replace(Transformer<>)`](javadoc/org/gradle/api/file/ConfigurableFileCollection.html#replace-org.gradle.api.Transformer-) for more details, including limitations.

Reviewer: ```suggestion
Refer to the Javadoc for [`Property.replace(Transformer<>)`](javadoc/org/gradle/api/provider/Property.html#replace-org.gradle.api.Transformer-) and [`ConfigurableFileCollection.replace(Transformer<>)`](javadoc/org/gradle/api/file/ConfigurableFileCollection.html#replace-org.gradle.api.Transformer-) for more details, including limitations.
```
